data/block_export_data.go

```diff
@@ -0,0 +1,29 @@
+package data
+
+import (
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+// RoundProvider is the interface which all data types sent to Exporters should implement
+type RoundProvider interface {
+	Round() uint64
+}
+
+// BlockData is provided to the Exporter on each round.
+type BlockData struct {
```

Comment:
It might be useful to standardize an "empty" function here that says if all pointers are nil then the block is empty and the next stage in the pipeline should not proceed.

Reply:
I added an `Empty` method. Let me know what you think...I don't think the Block should ever be nil--we had discussed just emptying out the set of transactions in a block as a way of still processing every Block, but stripping data that we don't want.

So I've ended up just attempting to return whether the length of the payset is greater than 0.

Reply:
I think we can create it as we need in the future.

Reply:
Another that we might want to consider is a "copy from another block" so we can facilitate copying an input block data so we can eventually return it as an output.

Reply:
I think data can be passed between and modified by each plugin as it proceeds through the processing steps. No need to copy blocks at every stage unless we really need to.